ultralytics/models/yolov10/predict.py

```diff
@@ -11,28 +11,33 @@ def postprocess(self, preds, img, orig_imgs):
 
         if isinstance(preds, (list, tuple)):
             preds = preds[0]
-        
-        if preds.shape[-1] == 6:
-            pass
-        else:
+
+        if preds.shape[-1] != 6:
             preds = preds.transpose(-1, -2)
-            bboxes, scores, labels = ops.v10postprocess(preds, self.args.max_det, preds.shape[-1]-4)
+            bboxes, scores, labels = ops.v10postprocess(
+                preds, self.args.max_det, preds.shape[-1] - 4
+            )
             bboxes = ops.xywh2xyxy(bboxes)
-            preds = torch.cat([bboxes, scores.unsqueeze(-1), labels.unsqueeze(-1)], dim=-1)
+            preds = torch.cat(
+                [bboxes, scores.unsqueeze(-1), labels.unsqueeze(-1)], dim=-1
+            )
 
         mask = preds[..., 4] > self.args.conf
 
-        b, _, c = preds.shape
-        preds = preds.view(-1, preds.shape[-1])[mask.view(-1)]
-        preds = preds.view(b, -1, c)
```

Comment:
Could we only replace these three lines with `preds = [p[mask[idx]] for idx, p in enumerate(preds)]`?